src/shared/install.h

```diff
@@ -74,7 +74,7 @@ enum UnitFileChangeType {
         UNIT_FILE_UNLINK,
         UNIT_FILE_IS_MASKED,
         _UNIT_FILE_CHANGE_TYPE_MAX,
-        _UNIT_FILE_CHANGE_TYPE_INVALID = -1
+        _UNIT_FILE_CHANGE_INVALID = INT_MIN
 };
```

Comment:
We actually only need to translate non-negative values, because we pass only such over the bus. I now modified the patch to share the message generation code path between bus and non-bus paths; it is moved to install.c.

I added a bunch of checks in various places, so the code should be gracefully handle the case where an older systemctl client is dealing with newer pid1.